mlflow/pipelines/steps/automl/flaml.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from typing import Dict, Any, Tuple
+
+import pandas as pd
+from sklearn.base import BaseEstimator
+
+import mlflow
+from mlflow import MlflowException
+from mlflow.models.evaluation.default_evaluator import _get_regressor_metrics
+from mlflow.pipelines.utils.metrics import _load_one_custom_metric_function, PipelineMetric
+
+_logger = logging.getLogger(__name__)
+
+AUTOML_DEFAULT_TIME_BUDGET = 10
+MLFLOW_TO_FLAML_METRICS = {
+    "mean_absolute_error": "mae",
+    "mean_squared_error": "mse",
+    "root_mean_squared_error": "rmse",
+    "r2_score": "r2",
+    "mean_absolute_percentage_error": "mape",
+}
+
+
+def get_estimator(
+    X,
+    y,
+    step_config: Dict[str, Any],
+    pipeline_root: str,
+    evaluation_metrics: Dict[str, PipelineMetric],
+    primary_metric: str,
+) -> Tuple[BaseEstimator, Dict[str, Any]]:
+    return _create_model_automl(
+        X, y, step_config, pipeline_root, evaluation_metrics, primary_metric
+    )
+
+
+def _create_custom_metric_flaml(metric_name: str, coeff: int, custom_func: callable) -> callable:
+    def add_suffix(metrics: Dict[str, float], suffix: str) -> Dict[str, float]:
+        return {f"{k}_{suffix}": v for k, v in metrics.items()}
+
+    def calc_metric(X, y, estimator) -> Dict[str, float]:
+        y_pred = estimator.predict(X)
+        builtin_metrics = _get_regressor_metrics(y, y_pred, sample_weights=None)
+        res_df = pd.DataFrame()
+        res_df["prediction"] = y_pred
+        res_df["target"] = y.values
+        return custom_func(res_df, builtin_metrics)
+
+    # pylint: disable=keyword-arg-before-vararg
+    # pylint: disable=unused-argument
```

Comment:
Why do we need to do this? Can we just not pass `labels`, `weight_val`, and `weight_train`?